src/main/java/com/aws/greengrass/security/provider/pkcs11/PKCS11CryptoKeyService.java

```diff
@@ -138,10 +138,29 @@ public PKCS11CryptoKeyService(Topics topics, SecurityService securityService) {
     protected void install() throws InterruptedException {
         try {
             super.install();
-            this.config.lookup(CONFIGURATION_CONFIG_KEY, NAME_TOPIC).subscribe(this::updateName);
-            this.config.lookup(CONFIGURATION_CONFIG_KEY, LIBRARY_TOPIC).subscribe(this::updateLibrary);
-            this.config.lookup(CONFIGURATION_CONFIG_KEY, SLOT_ID_TOPIC).subscribe(this::updateSlotId);
-            this.config.lookup(CONFIGURATION_CONFIG_KEY, USER_PIN_TOPIC).subscribe(this::updateUserPin);
+            this.config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe((whatHappened, node) -> {
```

Comment:
I believe this will create a new subscription every time `install` is called, since lambda will refer to different object each time.

Reply:
Good catch. Thanks!
Fixed it by moving the lambda to be a field of the class.

Reply:
Glad we're already feeding that knowledge back in to improve our code :D